### PR TITLE
feat(shorebird_cli): add ability to parse version number from plist

### PR DESCRIPTION
**Pull request by bryanoltman**
## Description

Adds support for deriving an app's version number from an ipa file.

## Type of Change

- [x] ✨ New feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code refactor
- [ ] ✅ Build configuration change
- [ ] 📝 Documentation
- [ ] 🗑️ Chore

### CODE REVIEW

File: packages/shorebird_cli/lib/src/archive_analysis/ipa.dart [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  /// The version number of the IPA, as derived from the app's Info.plist.
+  String get versionNumber {
+    final plist = _getPlist();
+    final releaseVersion = plist[releaseVersionKey] as String;
+    final buildNumber = plist[buildNumberKey] as String;
+    return '$releaseVersion+$buildNumber';
+  }
+
+  Map<dynamic, dynamic> _getPlist() {
+    final plistPathRegex = RegExp(r'Payload/[\w]+.app/Info.plist');
```
Comment (eseidel) [inline packages/shorebird_cli/lib/src/archive_analysis/ipa.dart:38]: No clue if this is the right path, but sure 🤷

File: packages/shorebird_cli/pubspec.yaml [hunk 2]
```diff
@@ -25,6 +25,7 @@ dependencies:
   meta: ^1.9.0
   path: ^1.8.3
   platform: ^3.1.0
+  plist_parser: ^0.0.9
```
Comment (eseidel) [inline packages/shorebird_cli/pubspec.yaml:28]: We're going to need to do a license review of our dependencies at some point.  lgtm though.

**Review by eseidel — APPROVED** [review]

File: packages/shorebird_cli/pubspec.yaml (hunk 2 above)
Comment (bryanoltman) [reply packages/shorebird_cli/pubspec.yaml:28]: Yep, agreed. This particular package has the MIT license, fwiw https://github.com/dirablue/plist_parser/blob/master/LICENSE

**Review by felangel — APPROVED** [review]

**Comment by bryanoltman** [comment]
Closing, as I don't think this is functionality we need at the moment (it seems neither releasing nor patching will involve building an ipa file)

**Pull request closed without merging**